srcpkgs/avr-gdb/template

```diff
@@ -1,79 +1,38 @@
 # Template file for 'avr-gdb'
 pkgname=avr-gdb
 version=10.1
-revision=1
+revision=2
 wrksrc=gdb-${version}
 build_style=gnu-configure
-pycompile_dirs="/usr/share/gdb/python"
-configure_args="\
-	--target=avr \
-	--program-prefix=avr- \
-	--disable-werror \
-	--disable-nls \
-	--with-system-readline \
-	--with-system-gdbinit=/etc/gdb/gdbinit \
-	--with-system-zlib $(vopt_enable gdbserver) \
-	$(vopt_if static 'CFLAGS=-static CXXFLAGS=-static LDFLAGS=-static') \
-	$(vopt_if python --with-python=/usr/bin/python3)"
-hostmakedepends="texinfo perl $(vopt_if python python3-devel)"
-makedepends="ncurses-devel zlib-devel readline-devel expat-devel
+configure_args="--disable-werror --disable-nls --with-system-readline
+ --with-system-gdbinit=/etc/gdb/gdbinit --with-system-zlib
+ $(vopt_with guile) $(vopt_if python --with-python=/usr/bin/python3)
+ --target=avr"
+hostmakedepends="pkg-config texinfo $(vopt_if python python3-devel) $(vopt_if guile guile)"
+makedepends="expat-devel ncurses-devel readline-devel zlib-devel $(vopt_if guile guile-devel)
  $(vopt_if python 'python3-devel gettext-devel')"
-depends=gdb
+depends="gdb-common"
 short_desc="GNU Debugger for AVR"
 maintainer="Artur Sinila <opensource@logarithmus.dev>"
 license="GPL-3.0-or-later"
-homepage="https://www.gnu.org/software/gdb/"
+homepage="https://www.gnu.org/software/gdb"
 distfiles="${GNU_SITE}/gdb/gdb-${version}.tar.xz"
 checksum=f82f1eceeec14a3afa2de8d9b0d3c91d5a3820e23e0a01bbb70ef9f0276b62c0
-python_version=3
+make_check=no  # See gdb/template.
 replaces="cross-avr-gdb>=0"
 
 if [ "${CROSS_BUILD}" ]; then
 	# Make python3.x detection work in cross builds
 	CFLAGS="-I${XBPS_CROSS_BASE}/${py3_inc}"
 	CXXFLAGS="-I${XBPS_CROSS_BASE}/${py3_inc}"
+	makedepends+=" libatomic_ops-devel" # guile
```

Comment:
`makedepends` should be unconditional, most of the time. Are you sure this isn't because you were cross building for a target like `armv6l`, which fits under our `XBPS_TARGET_NO_ATOMIC8` conditional?

Anyway, if this is guile related it should also be hidden behind `vopt_if`

Reply:
Here is what I get without `libatomic_ops-devel`:

```console
$ ./xbps-src -a aarch64-musl -o guile pkg gdb
…
configure: error: in `/builddir/gdb-10.1/gdb':
configure: error: linking guile version guile-2.2 test program failed
See `config.log' for more details
make[1]: *** [Makefile:9734: configure-gdb] Error 1
make[1]: *** Waiting for unfinished jobs....

$ grep -C1 'cannot find -latomic_ops' masterdir/builddir/gdb-10.1/gdb/config.log
configure:11330: aarch64-linux-musl-gcc -o conftest -fstack-clash-protection -D_FORTIFY_SOURCE=2 -O2 -pipe -march=armv8-a -I/usr/aarch64-linux-musl/usr/include/python3.9 -g -I/usr/aarch64-linux-musl/usr/include    -I/usr/aarch64-linux-musl/usr/include -I/usr/aarch64-linux-musl/usr/include/guile/2.2  -Wl,-z,relro -Wl,-z,now -Wl,--as-needed    -L/usr/aarch64-linux-musl/usr/lib conftest.c -lncursesw -lm  -lguile-2.2 -latomic_ops -lgc >&5
/usr/lib/gcc/aarch64-linux-musl/10.2.1/../../../../aarch64-linux-musl/bin/ld: cannot find -latomic_ops
collect2: error: ld returned 1 exit status
--
configure:11339: aarch64-linux-musl-gcc -o conftest -fstack-clash-protection -D_FORTIFY_SOURCE=2 -O2 -pipe -march=armv8-a -I/usr/aarch64-linux-musl/usr/include/python3.9 -g -I/usr/aarch64-linux-musl/usr/include    -I/usr/aarch64-linux-musl/usr/include -I/usr/aarch64-linux-musl/usr/include/guile/2.2  -Wl,-z,relro -Wl,-z,now -Wl,--as-needed    -L/usr/aarch64-linux-musl/usr/lib conftest.c -lncursesw -lm  -lguile-2.2 -latomic_ops -lgc >&5
/usr/lib/gcc/aarch64-linux-musl/10.2.1/../../../../aarch64-linux-musl/bin/ld: cannot find -latomic_ops
collect2: error: ld returned 1 exit status
```
```

Reply:
So, pkgconfig on armv6l says:

```
Libs: -L${libdir} -lguile-2.2 -latomic_ops -lgc
```

I'd suggest adding `libatomic_ops-devel` as a dependency of `guile-devel` instead :)

Reply:
Ping

Reply:
Do I understand correctly that you suggest the following patch:

File: srcpkgs/guile/template
```diff
@@ -40,7 +40,7 @@ libguile_package() {
 
 guile-devel_package() {
 	short_desc+=" - development files"
-	depends="gmp-devel gc-devel guile>=${version}_${revision}"
+	depends="gmp-devel gc-devel guile>=${version}_${revision} libatomic_ops-devel"
 	pkg_install() {
 		vmove usr/bin/guile-config
 		vmove usr/include
```

Reply:
Exactly :)

Reply:
Oh, no, there is the same bug as in #27320.

https://github.com/void-linux/void-packages/pull/29382/checks?check_run_id=2478214035
```
…
FAIL: numbers.test: Number-theoretic division: euclidean/: mixed types: (130.0 10/7)
FAIL: numbers.test: Number-theoretic division: euclidean/: mixed types: (130.0 -10/7)
FAIL: numbers.test: Number-theoretic division: floor/: mixed types: (130.0 10/7)
FAIL: numbers.test: Number-theoretic division: floor/: mixed types: (-130.0 -10/7)
FAIL: numbers.test: Number-theoretic division: ceiling/: mixed types: (130.0 -10/7)
FAIL: numbers.test: Number-theoretic division: ceiling/: mixed types: (-130.0 10/7)
FAIL: numbers.test: Number-theoretic division: truncate/: mixed types: (130.0 10/7)
FAIL: numbers.test: Number-theoretic division: truncate/: mixed types: (130.0 -10/7)
FAIL: numbers.test: Number-theoretic division: truncate/: mixed types: (-130.0 10/7)
FAIL: numbers.test: Number-theoretic division: truncate/: mixed types: (-130.0 -10/7)
…
```